resolver/caching_resolver.go

```diff
@@ -197,6 +202,24 @@ func (r *CachingResolver) Resolve(request *model.Request) (response *model.Respo
 	return response, err
 }
 
+func (r *CachingResolver) getFromCache(key string) (*dns.Msg, time.Duration) {
+	val, ttl := r.resultCache.Get(key)
+	if val != nil {
+		res := new(dns.Msg)
+
+		err := res.Unpack(*val)
+		if err != nil {
+			log.Log().Error("can't unpack cached entry. Cache malformed?", err)
+
+			return nil, 0
+		}
+
+		return res, ttl
+	}
+
+	return nil, 0
```

Comment:
Change to early return to reduce right-drift, and replaced `log.Log` with `r.log`:
```suggestion
	val, ttl := r.resultCache.Get(key)
	if val == nil {
		return nil, 0
	}

	res := new(dns.Msg)

	err := res.Unpack(*val)
	if err != nil {
		r.log().Error("can't unpack cached entry. Cache malformed?", err)

		return nil, 0
	}

	return res, ttl
```